### DISCUSSION TITLE
Antenna Efficiency Including Mismatch

### DISCUSSION THREAD

**Discussion by FolkSong** [thliebig/openEMS-Project#179] [Q&A Usage] [answered]
Hi, I'm trying out the inverted_f.m example file, and I wanted to calculate efficiency at a given frequency rather than only at resonance. The efficiency calculation in the script is:
`disp( ['efficiency: nu_rad = ' num2str(100*nf2ff.Prad./real(P_in(f_res_ind))) ' %']);`

To calculate radiated power I added a second frequency into the call to CalcNF2FF so that that nf2ff.Prad will be a 2-element vector. I can use the first element of nf2ff.Prad for the resonance efficiency calculation and the second for the fixed frequency calc.
`nf2ff = CalcNF2FF(nf2ff, Sim_Path, [f_res f_test], thetaRange*pi/180, phiRange*pi/180,'Verbose',1,'Outfile','3D_Pattern.h5');`

Then for the input power I can pass the index of the frequency of interest, f_test_ind. So my calculation will be:
`disp( ['Efficiency @ f_test: ' num2str(100*nf2ff.Prad(2)./real(P_in(f_test_ind))) ' %']);`

The problem is I get ~99% efficiency no matter what frequency I specify - that can't be right since the input match is poor at the test frequency (say 2 GHz). I notice that P_in itself varies across frequency and peaks at the resonance frequency, so it seems this is the power that gets into the antenna. How can I find the input power to the port, so I can calculate the true efficiency including mismatch loss?

And a follow-up question, apologies if I'm ignorant of basic theory: Does it make sense that the antenna is 99% efficient at any frequency if mismatch is ignored? Doesn't the structure of the antenna affect radiation efficiency itself?

**Comment by VolkerMuehlhaus** [comment, marked as answer]
> Does it make sense that the antenna is 99% efficient at any frequency if mismatch is ignored? Doesn't the structure of the antenna affect radiation efficiency itself?

@FolkSong Mismatch will reduce the **power accepted by the antenna**, so that it is less than **available power for matched load**. However, mismatch has no effect on radiation efficiency, because that is calculated as **total radiated powe**r divided by **power accepted by the antenna**. 

Radiation efficiency is less than 100% if there is **loss** (dissipation) inside the antenna, e.g. conductor loss or substrate loss. There is some substrate loss included in the PIFA example, but that is rather small with tand=0.001.

### Testcase to show effect of mismatch and loss

To test and show the various calculations for directivity, gain and realized gain I have created a patch antenna example. 

This antenna has ~50% radiation efficiency and also some mismatch loss. The model below is the Matlab/Octave version of the [patch antenna testcase for Python that I posted in another thread](https://github.com/thliebig/openEMS-Project/discussions/164#discussioncomment-7871184).

For simplicity, I only used the plotFFdB() for the pattern, which shows directivity only. The gain and realized gain calculations are only done for the peak value, and not shown for the angle sweep


![pattern_testcase_patch](https://github.com/thliebig/openEMS-Project/assets/47922981/d1cdd3b5-1646-4777-8f6a-371533845999)



```
%
% based on openEMS EXAMPLE / antennas / patch antenna
% but different substrate with high loss and poor matching,
% to see the effect of radiation effieincy and mismatch
%


close all
clear
clc

%% switches & options...
postprocessing_only = 0;
openEMS_opts = '';

%% setup the simulation
physical_constants;
unit = 1e-3; % all length in mm

% width in x-direction
% length in y-direction
% main radiation in z-direction
patch.width  = 29;
patch.length = 38;

substrate.epsR   = 4.0;
substrate.tand   = 0.02;
substrate.kappa  = substrate.tand * 2*pi*2.45e9 * EPS0*substrate.epsR;
substrate.width  = 60;
substrate.length = 60;
substrate.thickness = 1.524;
substrate.cells = 4;

feed.pos = -5;
feed.width = 1;
feed.R = 50; % feed resistance

% size of the simulation box
SimBox = [100 100 25];

%% prepare simulation folder
Sim_Path = 'tmp';
Sim_CSX = 'patch_ant.xml';
if (postprocessing_only==0)
    [status, message, messageid] = rmdir( Sim_Path, 's' ); % clear previous directory
    [status, message, messageid] = mkdir( Sim_Path ); % create empty simulation folder
end

%% setup FDTD parameter & excitation function
max_timesteps = 30000;
min_decrement = 1e-5; % equivalent to -50 dB
f0 = 2.45e9; % center frequency
fc = 1e9; % 20 dB corner frequency

# bandwidth for frequency plots
plot_bw = 0.2e9; # bandwidth for plots (less than excitation bandwidth)

FDTD = InitFDTD( 'NrTS', max_timesteps, 'EndCriteria', min_decrement );
FDTD = SetGaussExcite( FDTD, f0, fc );
BC = {'PML_8' 'PML_8' 'PML_8' 'PML_8' 'PML_8' 'PML_8'}; % use pml instead of mur
FDTD = SetBoundaryCond( FDTD, BC );

%% setup CSXCAD geometry & mesh
% currently, openEMS cannot automatically generate a mesh
max_res = c0 / (f0+fc) / unit / 20; % cell size: lambda/20
CSX = InitCSX();
mesh.x = [-SimBox(1)/2 SimBox(1)/2 -substrate.width/2 substrate.width/2 feed.pos];
% add patch mesh with 2/3 - 1/3 rule
mesh.x = [mesh.x -patch.width/2-max_res/2*0.66 -patch.width/2+max_res/2*0.33 patch.width/2+max_res/2*0.66 patch.width/2-max_res/2*0.33];
mesh.x = SmoothMeshLines( mesh.x, max_res, 1.4); % create a smooth mesh between specified mesh lines
mesh.y = [-SimBox(2)/2 SimBox(2)/2 -substrate.length/2 substrate.length/2 -feed.width/2 feed.width/2];
% add patch mesh with 2/3 - 1/3 rule
mesh.y = [mesh.y -patch.length/2-max_res/2*0.66 -patch.length/2+max_res/2*0.33 patch.length/2+max_res/2*0.66 patch.length/2-max_res/2*0.33];
mesh.y = SmoothMeshLines( mesh.y, max_res, 1.4 );
mesh.z = [-SimBox(3)/2 linspace(0,substrate.thickness,substrate.cells) SimBox(3) ];
mesh.z = SmoothMeshLines( mesh.z, max_res, 1.4 );
mesh = AddPML( mesh, [8 8 8 8 8 8] ); % add equidistant cells (air around the structure)
CSX = DefineRectGrid( CSX, unit, mesh );

%% create patch
CSX = AddMetal( CSX, 'patch' ); % create a perfect electric conductor (PEC)
start = [-patch.width/2 -patch.length/2 substrate.thickness];
stop  = [ patch.width/2  patch.length/2 substrate.thickness];
CSX = AddBox(CSX,'patch',10,start,stop);

%% create substrate
CSX = AddMaterial( CSX, 'substrate' );
CSX = SetMaterialProperty( CSX, 'substrate', 'Epsilon', substrate.epsR, 'Kappa', substrate.kappa );
start = [-substrate.width/2 -substrate.length/2 0];
stop  = [ substrate.width/2  substrate.length/2 substrate.thickness];
CSX = AddBox( CSX, 'substrate', 0, start, stop );

%% create ground (same size as substrate)
CSX = AddMetal( CSX, 'gnd' ); % create a perfect electric conductor (PEC)
start(3)=0;
stop(3) =0;
CSX = AddBox(CSX,'gnd',10,start,stop);

%% apply the excitation & resist as a current source
start = [feed.pos-.1 -feed.width/2 0];
stop  = [feed.pos+.1 +feed.width/2 substrate.thickness];
[CSX port]  = AddLumpedPort(CSX, 5 ,1 ,feed.R, start, stop, [0 0 1], true);


%%nf2ff calc
[CSX nf2ff] = CreateNF2FFBox(CSX, 'nf2ff', -SimBox/2, SimBox/2);

if (postprocessing_only==0)
    %% write openEMS compatible xml-file
    WriteOpenEMS( [Sim_Path '/' Sim_CSX], FDTD, CSX );

    %% show the structure
    CSXGeomPlot( [Sim_Path '/' Sim_CSX] );

    %% run openEMS
    RunOpenEMS( Sim_Path, Sim_CSX, openEMS_opts );
end

%% postprocessing & do the plots
freq = linspace( f0-plot_bw/2, f0+plot_bw/2, 501 );
port = calcPort(port, Sim_Path, freq);

Zin = port.uf.tot ./ port.if.tot;
s11 = port.uf.ref ./ port.uf.inc;


% plot feed point impedance
figure
plot( freq/1e6, real(Zin), 'k-', 'Linewidth', 2 );
hold on
grid on
plot( freq/1e6, imag(Zin), 'r--', 'Linewidth', 2 );
title( 'feed point impedance' );
xlabel( 'frequency f / MHz' );
ylabel( 'impedance Z_{in} / Ohm' );
legend( 'real', 'imag' );

% plot reflection coefficient S11
figure
plot( freq/1e6, 20*log10(abs(s11)), 'k-', 'Linewidth', 2 );
grid on
title( 'reflection coefficient S_{11}' );
xlabel( 'frequency f / MHz' );
ylabel( 'reflection coefficient |S_{11}|' );

drawnow

% Antenna pattern calculations at resonance

f_res_ind = find(s11==min(s11));
f_res = freq(f_res_ind);

% calculate the far field at phi=0 degrees and at phi=90 degrees
thetaRange = (0:2:359) - 180;
phiRange = [0 90];
disp( 'calculating far field at phi=[0 90] deg...' );
nf2ff = CalcNF2FF(nf2ff, Sim_Path, f_res, thetaRange*pi/180, phiRange*pi/180);

% get total radiated power at nf2ff frequency (here: resonance)
Prad_total = nf2ff.Prad;

% port power wer calculations
%   port.P_inc              incoming power
%   port.P_ref              reflected power
%   port.P_acc              accepted power (incoming minus reflected,
%                           may be negative for passive ports)

% get accepted power into port (this can be reduced by mismatch)
P_accepted = port.P_acc(f_res_ind);
% get incident power
P_incident = port.P_inc(f_res_ind) ;

% loss from mismatch
mismatch_loss = P_accepted / P_incident;
mismatch_loss_dB = 10*log10(mismatch_loss);

% calculate radiation efficiency (lossless antenna is 1, regardless of input matching)
% radiation efficiency in dB is negative for lossy antenna)
radiation_efficiency = Prad_total / P_accepted;
radiation_efficiency_dB = 10*log10(radiation_efficiency);

% directivity at nf2ff frequency (peak value)
Directivity_max_dB = 10*log10(nf2ff.Dmax);
Gain_max_dB = Directivity_max_dB + radiation_efficiency_dB;
Realized_Gain_max_dB = Gain_max_dB + mismatch_loss_dB;

% display power and directivity
disp( ['incident power: Pincident = ' num2str(P_incident) ' Watt']);
disp( ['accepted power: Paccepted = ' num2str(P_accepted) ' Watt']);
disp( ['radiated power: Prad = ' num2str(Prad_total) ' Watt']);
disp( ['efficiency: nu_rad = ' num2str(100*radiation_efficiency) ' %']);
disp( ['directivity: Dmax = ' num2str(Directivity_max_dB) ' dBi'] );
disp( ['gain       : Gmax = ' num2str(Gain_max_dB) ' dBi'] );
disp( ['realized gain : Gmax,realized = ' num2str(Realized_Gain_max_dB) ' dBi'] );

% display phi
figure
plotFFdB(nf2ff,'xaxis','theta','param',[1 2]);
drawnow

```

**Reply by FolkSong** [reply]
Thanks very much @VolkerMuehlhaus, that makes perfect sense that if the substrate is too ideal there will be no loss. And thanks also for clarifying radiation efficiency versus total efficiency. 

I'll add these considerations to my simulation and carry on.

**Comment by FolkSong** [comment]
One follow up question if you don't mind, on general antenna design: Is there any downside to just optimizing the design for radiation efficiency and then using discrete components to correct mismatch?

**Reply by VolkerMuehlhaus** [reply]
@FolkSong The matching network will work only for a limited bandwidth. LC matching using SMD elements also introduces additional losses because the inductors have limited Q factor. So you should design for a "useful" impedance at the antenna, not too far away from your target value. Rule of thumb: The more impedance transformation you need, the higher the losses in the matching network, and the lower the matching network bandwidth.

There is a commercial tool for antenna matching from Finlad (optennilab) which optimizes for radiated power instead of return loss, and includes all the losses in the matching network in that calculation. I think that is a very smart approach.

For a given antenna topology, in most cases you can't really change radiation efficiency. Except of course, you switch to a lower loss substrate etc, but then for the better substrate you can optimize for return loss again.
